fre/cmor/cmor_mixer.py

```diff
@@ -1,408 +1,596 @@
-    for name, variable in ds.variables.items():
-        if name == gfdl_var:
-            dims = variable.dimensions
-            for dim in dims:
-                if ds[dim].axis and ds[dim].axis == "Z":
-                    vert_dim = dim
-    print(f"(netcdf_var) Vertical dimension: {vert_dim}")
+def check_dataset_for_ocean_grid(ds):
+    '''
+    checks netCDF4.Dataset ds for ocean grid origin, and throws an error if it finds one. accepts
+    one argument. this function has no return.
+        ds: netCDF4.Dataset object containing variables with associated dimensional information.
+    '''
+    #print(f'(check_dataset_for_ocean_grid) {ds}')
+    #print(f'(check_dataset_for_ocean_grid) {ds.variables}')
+    #print(f'(check_dataset_for_ocean_grid) {ds.variables.keys()}')
+    if "xh" in list(ds.variables.keys()):
+        raise NotImplementedError(
+            "'xh' found in var_list. ocean grid req'd but not yet unimplemented. stop.")
 
-    # initialize CMOR
-    cmor.setup()
+def get_vertical_dimension(ds,target_var):
+    '''
+    determines the vertical dimensionality of target_var within netCDF4 Dataset ds. accepts two
+    arguments and returns an object represnting the vertical dimensions assoc with the target_var.
+        ds: netCDF4.Dataset object containing variables with associated dimensional information.
+        target_var: string, representating a variable contained within the netCDF4.Dataset ds
 
-    # read experiment configuration file
-    cmor.dataset_json(cmip_input_json)
-    print(f"(netcdf_var) cmip_input_json = {cmip_input_json}")
-    print(f"(netcdf_var) cmor_table_vars_file = {cmor_table_vars_file}")
+    '''
+    vert_dim = 0
+    for name, variable in ds.variables.items():
+        # not the var we are looking for? move on.
+        if name != target_var:
+            continue
+        dims = variable.dimensions
+        for dim in dims:
+            # if it is not a vertical axis, move on.
+            if not (ds[dim].axis and ds[dim].axis == "Z"):
+                continue
+            vert_dim = dim
+    return vert_dim
+
+
+def create_tmp_dir(outdir):
+    '''
+    creates a tmp_dir based on targeted output directory root. returns the name of the tmp dir.
+    accepts one argument:
+        outdir: string, representing the final output directory root for the cmor modules netcdf
+                file output. tmp_dir will be slightly different depending on the output directory
+                targeted
+    '''
+    print(f"(cmorize_target_var_files) outdir = {outdir}")
+    tmp_dir = None
+    if any( [ outdir == "/local2",
+              outdir.find("/work") != -1,
+              outdir.find("/net" ) != -1 ] ):
+        print(f'(cmorize_target_var_files) using /local /work /net ( tmp_dir = {outdir}/ )')
+        tmp_dir = "{outdir}/"
+    else:
+        print('(cmorize_target_var_files) NOT using /local /work /net (tmp_dir = outdir/tmp/ )')
+        tmp_dir = f"{outdir}/tmp/"
```

Comment:
I believe you that this is needed for some reason, but it's odd. This is the logic, right?

If the output directory is on /work or /net2, then set tmpdir to that directory.

If the output directory is somewhere else, set tmpdir to the current directory.

Reply:
I am not entirely convinced this is needed for any reason. If there exists one, it's clear to someone else, not me. I only attempted only to preserve this directory creation logic as I originally encountered it.

Your read of what's happening isn't exaaaaactly right, to split hairs...
- if the output directory is '/local2', or it's on '/work', or '/net', then use the output directory specified
- otherwise, create a new directory within outdir called 'tmp' and use that.

TLDR- We can put in whatever file movement logic here we want! I am not entirely sure whats best for gfdl/ppan, but if someone can tell me, i'll implement it.

Reply:
Alright, I see.  Next time we are confused by the different tmpdir behavior tho let's make them the same. It looks like they are really the same anyway, with a confusing twist on the /local2 /work thing that doesn't even matter.